security/design/oauth2.go

```diff
@@ -0,0 +1,50 @@
+package design
+
+import (
+	. "github.com/goadesign/goa/design"
+	. "github.com/goadesign/goa/design/apidsl"
+)
+
+// OAuth defines a security scheme using OAuth2. The scheme uses the standard "access code" flow
+// where the application must first retrieve an authorization code to request for the refresn and
+// access tokens.
+//
+var OAuth2 = OAuth2Security("oauth2", func() {
+	// /oauth2/code is the path to the action that produces access codes
+	// /oauth2/token is the path to the action that produces refresh and access tokens from
+	// access codes and existing refresh tokens.
+	AccessCodeFlow("/oauth2/code", "/oauth2/token")
+	// All secured API requests must carry the "api" scope.
+	Scope("api")
```

Comment:
And in one case have it required (desc) ?  I think that was the behavior
already. But I'm not against having two funcs.. Would clear things up
indeed.

Le jeu. 5 mai 2016 23:51, Raphaël Simon <notifications@github.com> a écrit :

> In security/design/oauth2.go
> <https://github.com/goadesign/examples/pull/4#discussion_r62286224>:
>
> > +import (
> > +	. "github.com/goadesign/goa/design"
> > +	. "github.com/goadesign/goa/design/apidsl"
> > +)
> > +
> > +// OAuth defines a security scheme using OAuth2. The scheme uses the standard "access code" flow
> > +// where the application must first retrieve an authorization code to request for the refresn and
> > +// access tokens.
> > +//
> > +var OAuth2 = OAuth2Security("oauth2", func() {
> > +	// /oauth2/code is the path to the action that produces access codes
> > +	// /oauth2/token is the path to the action that produces refresh and access tokens from
> > +	// access codes and existing refresh tokens.
> > +	AccessCodeFlow("/oauth2/code", "/oauth2/token")
> > +	// All secured API requests must carry the "api" scope.
> > +	Scope("api")
>
> It just seems strange to have the field "Description" be required. It's
> not anywhere else in the DSL. I think it would be good to have different
> names for the different use of Scope (define vs. use). It's a little bit
> confusing as. Maybe Scope and RequireScope ?
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/goadesign/examples/pull/4/files/a8a6222055b0636071a616d67677cb14040e1176#r62286224>
>